lib/puppet/parser/functions/pick.rb

```diff
@@ -15,9 +15,22 @@ module Puppet::Parser::Functions
     called 'jenkins_version' (note that parameters set in the Puppet Dashboard/
     Enterprise Console are brought into Puppet as top-scope variables), and,
     failing that, will use a default value of 1.449.
+
+    If you have `strict_variables` turned on, then wrap your variable in single
+    quotes to prevent interpolation and this function will check to see if that
+    variable exists.
+
+      $real_jenkins_version = pick('$::jenkins_version', '1.449')
+
 DOC
              ) do |args|
-    args = args.compact
+
+    # look up the values of any strings that look like '$variables'
+    args.map! do |item|
+      next unless item.is_a? String
+      item.start_with?('$') ? function_getvar([item.slice(1..-1)]) : item
```

Comment:
Use `call_function` instead of calling with `function_xxx` as that only works for 3.x functions.